app/services/sp_return_url_resolver.rb

```diff
@@ -0,0 +1,45 @@
+class SpReturnUrlResolver
+  attr_reader :sp, :sp_request_url, :view_context
+
+  def initialize(sp:, sp_request_url:)
+    @sp = sp
```

Comment:
WDYT of expanding this to be `service_provider` instead of just `sp`?

Reply:
Good call